### ISSUE TITLE
Update allowances for the area element without href

### ISSUE THREAD

**Issue by scottaohara** [validator/validator#1249]
The *`area` element without `href`* presently does not allow any role to be specified by authors.  The [ARIA in HTML specification is updating this restriction](https://github.com/w3c/html-aria/pull/360) to allow authors to specify a `button` or `link` role. 
```html
<map name=...>
  <area shape=rect coords=... alt=accName role=button> <!-- pass -->
  <area shape=rect coords=... alt=accName role=link> <!-- pass -->
</map>
```

All other roles would continue to not be allowed on the the `area` element if it lacks an `href`.

As without the `href` the `area` would no longer be keyboard focusable, conformance checkers may surface a warning stating this.  However, this will not be a requirement from ARIA in HTML.

**Comment by stevefaulkner** [comment]
@sideshowbarker had a look at where/how this should be defined in code, couldn't work it out

**Comment by sideshowbarker** [comment]
Note that per the current HTML spec, the `href`-less markup snippet in the issue description will only ever be valid if the `alt` attribute is omitted. That’s because the relevant part of the HTML spec at https://html.spec.whatwg.org/multipage/image-maps.html#the-area-element:attr-area-alt-4 says this:

> If the [`area`](https://html.spec.whatwg.org/multipage/image-maps.html#the-area-element) element has no [`href`](https://html.spec.whatwg.org/multipage/links.html#attr-hyperlink-href) attribute, then the area represented by the element cannot be selected, and **the [`alt`](https://html.spec.whatwg.org/multipage/image-maps.html#attr-area-alt) attribute must be omitted.**

**Comment by sideshowbarker** [comment]
> As without the `href` the `area` would no longer be keyboard focusable, conformance checkers may surface a warning stating this. However, this will not be a requirement from ARIA in HTML.

Please raise a separate issue for that —  because it’s preferable to restrict this issue to just the change needed to bring the checker into conformance with the relevant normative requirements.

**Comment by sideshowbarker** [comment]
Incidentally, can the `alt` content from an `href`-less `area` element can still get exposed usefully to AT somehow? If it has a `button` or `link` role, maybe?

If so, then I guess somebody should raise an issue in the HTML issue tracker about that restriction at https://html.spec.whatwg.org/multipage/image-maps.html#the-area-element:attr-area-alt-4
